src/store/Disks.cc

```diff
@@ -363,17 +375,27 @@ Store::Disks::maxObjectSize() const
 }
 
 void
-Store::Disks::updateLimits()
+Store::Disks::configure()
 {
+    if (!Config.cacheSwap.swapDirs)
+        Controller::store_dirs_rebuilding = 0; // nothing to index
+
     largestMinimumObjectSize = -1;
     largestMaximumObjectSize = -1;
     secondLargestMaximumObjectSize = -1;
 
     for (int i = 0; i < Config.cacheSwap.n_configured; ++i) {
-        const auto &disk = Dir(i);
+        auto &disk = Dir(i);
+        if (disk.needsDiskStrand()) {
+            assert(InDaemonMode());
+            disk.disker = Config.workers + (++Config.cacheSwap.n_strands);
+        }
```

Comment:
:triangular_flag_on_post: I am concerned that I see no PR code that zeroes the always-incremented n_strands counter during _reconfiguration_. We used to reset that counter in configDoConfigure(). Please double check. If my concern is valid, please fix initialization, retest reconfigure, and add an XXX:

```suggestion
        // XXX: Do not pretend to support disk.disker changes during reconfiguration
        if (disk.needsDiskStrand()) {
            assert(InDaemonMode());
            disk.disker = Config.workers + (++Config.cacheSwap.n_strands);
        }
```

I do not think we can easily address that old XXX, so I do not recommend addressing it in _this_ PR.

Reply:
> is this about that after reconfiguration disk.disker may change but KidIdentifier remains the same, thus breaking methods where they are compared (e.g Store::Disk::active())?

Yes, although I was thinking about a somewhat more general problem: A reconfiguration might change needsDiskStrand() outcome -- there are no API guarantees that needsDiskStrand() result is sticky -- but a worker cannot start (or stop) using a disker in the middle of a transaction. I also doubt our code can handle a sudden n_strands increase. In other words, we probably do not support Disk::disker and cacheSwap.n_strands reconfiguration in general.

In the current code, Disk::disker is barely used, so specific problems may be minor or even non-existent, but, as we all know, code tends to change. This code dangerously mixes first-time configuration and reconfiguration code, sometimes on purpose and sometimes essentially accidentally. This mixture will continue to break code long-term, but fixing that problem is outside this PR scope. Your changes will just make that fix a bit easier (and will add more reminders about the need for such a fix).